tests/e2e/pages/aelin/create-direct-deal.js

```diff
@@ -0,0 +1,468 @@
+    }
+
+    if (step >= CreateDirectDealSteps.ExchangeRates) {
+      this.getInvestmentAmountToRaiseInput().type(fakeDirectDeal.investmentAmountToRaise)
+      this.getExchangeRateInput().type(fakeDirectDeal.exchangeRate)
+      this.nextStep()
+    }
+  }
+
+  assertNoSponsorFeeCheckBox(checked) {
+    return this.getNoSponsorFeeCheckBox().should(
+      'have.css',
+      'background-color',
+      checked ? 'rgb(130, 128, 255)' : 'rgba(255, 255, 255, 0.04)',
+    )
+  }
+
+  assertPrivacyRadioButton(privacy, checked) {
+    return this.getPrivacyRadioButton(privacy).should(
+      'have.css',
+      'background-color',
+      checked ? 'rgb(130, 128, 255)' : 'rgba(255, 255, 255, 0.04)',
+    )
+  }
+
+  assertPrivacyRadioButtons(privacy) {
+    switch (privacy) {
+      case DirectDealPrivacy.PUBLIC:
+        this.assertPrivacyRadioButton(DirectDealPrivacy.PUBLIC, true)
+        this.assertPrivacyRadioButton(DirectDealPrivacy.PRIVATE, false)
+        this.assertPrivacyRadioButton(DirectDealPrivacy.NFT, false)
+        break
+      case DirectDealPrivacy.PRIVATE:
+        this.assertPrivacyRadioButton(DirectDealPrivacy.PUBLIC, false)
+        this.assertPrivacyRadioButton(DirectDealPrivacy.PRIVATE, true)
+        this.assertPrivacyRadioButton(DirectDealPrivacy.NFT, false)
+        break
+      case DirectDealPrivacy.NFT:
+        this.assertPrivacyRadioButton(DirectDealPrivacy.PUBLIC, false)
+        this.assertPrivacyRadioButton(DirectDealPrivacy.PRIVATE, false)
+        this.assertPrivacyRadioButton(DirectDealPrivacy.NFT, true)
+        break
+      default:
+        this.assertPrivacyRadioButton(DirectDealPrivacy.PUBLIC, false)
+        this.assertPrivacyRadioButton(DirectDealPrivacy.PRIVATE, false)
+        this.assertPrivacyRadioButton(DirectDealPrivacy.NFT, false)
+    }
+  }
+
+  assertDealCappedRadioButton(yes, checked) {
+    return this.getDealCappedRadioButton(yes).should(
+      'have.css',
+      'background-color',
+      checked ? 'rgb(130, 128, 255)' : 'rgba(255, 255, 255, 0.04)',
+    )
+  }
+
+  assertDealCappedRadioButtons(yes) {
+    if (yes) {
+      this.assertDealCappedRadioButton(true, true)
+      this.assertDealCappedRadioButton(false, false)
+    } else {
+      this.assertDealCappedRadioButton(true, false)
+      this.assertDealCappedRadioButton(false, true)
+    }
+  }
+
+  assertDealHasMinimumRadioButton(yes, checked) {
+    return this.getDealHasMinimumRadioButton(yes).should(
+      'have.css',
+      'background-color',
+      checked ? 'rgb(130, 128, 255)' : 'rgba(255, 255, 255, 0.04)',
+    )
+  }
+
+  assertDealHasMinimumRadioButtons(yes) {
+    if (yes) {
+      this.assertDealHasMinimumRadioButton(true, true)
+      this.assertDealHasMinimumRadioButton(false, false)
+    } else {
+      this.assertDealHasMinimumRadioButton(true, false)
+      this.assertDealHasMinimumRadioButton(false, true)
+    }
+  }
+
+  assertDetails(fakeDirectDeal, step = CreateDirectDealSteps.VestingSchedule, expected) {
```

Comment:
I think that this is a bit overkill.. why dont just add the code for each test directly? 
You will be adding more lines/less clean, but you will end up removing unnecessary abstraction imo.
And again... we are mixing concerns here.

Reply:
Well, that's arguable... I see tests as a good source of requirements about how app should work, so readability is very helpful not only to write/fix them but also to onboard new contributors and discover use cases in general